src/tests/test_fields.py

```diff
@@ -26,3 +27,31 @@ def test_float():
         (long(5.7), 5L)
     ]:
         assert floatify(input) == expect
+
+def test_url():
+    """
+    Assure that url performs some basic validation
+    """
+    urlify = fields.UrlField().adapt
+
+    # positive tests
+    for input, expect in [
+        ('http://www.google.com', 'http://www.google.com'),
+        ('hTTp://www.google.com', 'hTTp://www.google.com'),
```

Comment:
I'd prefer that we lowercase scheme and hostname for normalization, but not path/query.